analyzers/src/SonarAnalyzer.Common/Rules/BooleanLiteralUnnecessaryBase.cs

```diff
@@ -55,6 +55,7 @@ protected enum ErrorLocation
         protected abstract SyntaxToken GetOperatorToken(TBinaryExpression binaryExpression);
         protected abstract bool IsTrueLiteralKind(SyntaxNode syntaxNode);
         protected abstract bool IsFalseLiteralKind(SyntaxNode syntaxNode);
+        protected abstract bool IsInsideTernaryWithThrowExpression(TBinaryExpression syntaxNode);
```

Comment:
For this I don't have a strong opinion so please feel free to ignore it if you don't agree with it: It might be a bit nicer if this would be a virtual method returning false by default so the VB part should not be forced to override a method that is specific to C#.